File: src/Array/python/haoxiang/MajorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        n = len(nums)
        newNum = {}
        if n == 1:
            return nums[0]
        for i in range(n):
            if newNum.has_key(nums[i]):
                if newNum[nums[i]] >= (n/2):
                    return nums[i]
                else:
                    newNum[nums[i]] += 1
            else:
                newNum[nums[i]] = 1

File: src/qiuyi/HashTable/MajorityElement.cpp
// Solution 1
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> count;
        for (auto i : nums)
            if (++count[i] > nums.size()/2)
                return i;
    }
};

// Solution 2
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candidate=0, cnt=0;
        for (auto i : nums)
            cnt == 0 ? candidate = i, cnt++ : i == candidate ? cnt++ : cnt--;
        return candidate;
    }
};